cmd/restic/cmd_check_test.go

```diff
@@ -163,3 +165,75 @@ func TestSelectNoRandomPacksByFileSize(t *testing.T) {
 	selectedPacks := selectRandomPacksByFileSize(testPacks, 10, 500)
 	rtest.Assert(t, len(selectedPacks) == 0, "Expected 0 selected packs")
 }
+
+func TestPrepareCheckCache(t *testing.T) {
+	// Create a temporary directory for the cache
+	tmpDirBase, err := os.MkdirTemp("", "test-prepare-check-cache-basepath")
+	if err != nil {
+		t.Fatalf("Failed to create temporary directory: %v", err)
+	}
+	defer os.RemoveAll(tmpDirBase)
+
+	testCases := []struct {
+		opts        CheckOptions
+		cacheExists bool
+		expectNoErr bool
+	}{
+		{CheckOptions{WithCache: true}, false, true}, // Shouldn't create temp directory
+		{CheckOptions{WithCache: false}, true, true}, // Should create temp directory
+	}
+
+	for _, testCase := range testCases {
+		t.Run("", func(t *testing.T) {
+			gopts := GlobalOptions{CacheDir: tmpDirBase}
+			cleanup := prepareCheckCache(testCase.opts, &gopts)
+
+			if !testCase.opts.WithCache {
+				// If using a temporary cache directory, the cache directory should exist
+				// listing all directories inside tmpDirBase (cacheDir)
+				// one directory should be tmpDir created by prepareCheckCache with 'restic-check-cache-' in path
+				files, err := os.ReadDir(tmpDirBase)
+				if err != nil {
+					t.Errorf("Failed to list cacheDir.")
+				}
+
+				found := false
+				for _, file := range files {
+					if file.IsDir() {
+						dirName := file.Name()
+						if strings.Contains(dirName, "restic-check-cache-") {
+							found = true
+						}
+					}
+				}
+				if !found {
+					t.Errorf("Expected temporary directory to exist, but it does not")
+				}
+			} else {
+				// If not using the cache, the temp directory should not exist
+				files, err := os.ReadDir(tmpDirBase)
+				if err != nil {
+					t.Errorf("Failed to list cacheDir.")
+				}
+				if len(files) != 0 {
```

Comment:
Please use
`rtest.Assert(t, len(files) == 0, "expected cache directory not to exist, but it does: %v", files)`. That also applies to the other `len(...) != 0` checks.

Reply:
Is this only for consistency @MichaelEischer , I'm curious if there are more benefits than ```len(...)``` checks?

Reply:
The main benefit is that `rtest.Assert` takes less space to write than the `if ... { t.Errof(...) }`. By now, there's also `rtest.Equals(t, 0, len(files), "description")` that automatically prints both values if there's a mismatch.